app/src/test/java/org/simple/clinic/sync/indicator/SyncIndicatorViewControllerTest.kt

```diff
@@ -54,19 +54,22 @@ class SyncIndicatorViewControllerTest {
 
   @Test
   @Parameters(method = "params for testing sync status indicator update")
-  fun `when sync result is updated, sync status indicator should change`(syncProgress: SyncProgress, expectedSyncState: SyncIndicatorState) {
+  fun `when sync result is updated, sync status indicator should change`(lastSyncState: LastSyncedState, expectedSyncState: SyncIndicatorState) {
     uiEvents.onNext(SyncIndicatorViewCreated)
-    syncProgressStream.onNext(Just(syncProgress))
+    lastSyncStateStream.onNext(lastSyncState)
 
     verify(indicator).updateState(expectedSyncState)
   }
 
   @Suppress("Unused")
   private fun `params for testing sync status indicator update`(): List<List<Any>> {
     return listOf(
-        listOf(SYNCING, Syncing),
-        listOf(SUCCESS, Synced),
-        listOf(FAILURE, SyncPending)
+        listOf(LastSyncedState(SYNCING), Syncing),
+        listOf(LastSyncedState(SUCCESS), Synced),
+        listOf(LastSyncedState(FAILURE), SyncPending),
+        listOf(LastSyncedState(SUCCESS, Instant.now(utcClock).minus(20, ChronoUnit.MINUTES)), SyncPending),
```

Comment:
Another reason why the 12 hour duration should be injected via a config object, is that this test is written under the assumption that the duration is 12 hours. If we change that duration in the future, we will also have to change these tests, which doesn't make sense.

Injecting it via a configuration will let us separate what we are testing from what the production configuration is.

Reply:
This will be done in the next PR.